components/skus/browser/rs/lib/src/storage/kv.rs

```diff
@@ -331,6 +331,32 @@ where
         store.set_state(&state)
     }
 
+    #[instrument]
+    async fn upsert_time_limited_v2_item_creds_request_id(
+        &self,
+        item_id: &str,
+        request_id: &str,
+    ) -> Result<(), InternalError> {
+        let mut store = self.get_store()?;
+        let mut state: KVState = store.get_state()?;
+
+        if state.credentials.is_none() {
+            state.credentials = Some(CredentialsState::default());
+        }
+
+        if let Some(credentials) = state.credentials.as_mut() {
+            // check and see if we already have this item initialized
+            if let Some(item_credentials) = credentials.items.get_mut(item_id) {
+                // overwrite the request_id with the new value
+                if let Credentials::TimeLimitedV2(item_credentials) = item_credentials {
+                    item_credentials.request_id = Some(request_id.to_string());
+                }
```

Comment:
Wouldn't it be better to return an error if this is called with something besides as `Credentials::TimeLimitedV2` similar to `get_time_limited_v2_creds`?

Reply:
that makes sense to me